thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/DataAvailabilityTaskScheduler.java

```diff
@@ -0,0 +1,229 @@
+    List<DetectionConfigDTO> detectionConfigs = detectionConfigManager.findAll();
+    for (DetectionConfigDTO detectionConfig : detectionConfigs) {
+      if (!detectionConfig.isActive()) continue; // skip inactive detection
+      if (detectionConfig.getLixFlags().stream().filter(x -> x.equals(LIX_FLAG_VALUE)).count() < 1) continue;
+      List<String> metricUrns = DetectionConfigFormatter
+          .extractMetricUrnsFromProperties(detectionConfig.getProperties());
+      if (metricUrns.size() < 1) {
+        LOG.error("Empty metricUrn for detection {}", detectionConfig.getId());
+        continue;
+      }
+      Set<String> datasets = new HashSet<>();
+      for (String urn : metricUrns) {
+        MetricEntity me = MetricEntity.fromURN(urn);
+        if (!metricCache.containsKey(me.getId())) {
+          MetricConfigDTO metricConfig = metricConfigManager.findById(me.getId());
+          if (metricConfig == null) {
+            LOG.warn("Found null value for metric: " + me.getId());
+            metricCache.put(me.getId(), null);
+            continue;
+          }
+          if (metricConfig.isDerived()) {
+            MetricExpression metricExpression = ThirdEyeUtils.getMetricExpressionFromMetricConfig(metricConfig);
+            List<MetricFunction> functions = metricExpression.computeMetricFunctions();
+            functions.forEach(f -> datasets.add(f.getDataset()));
+          } else {
+            datasets.add(metricConfig.getDataset());
+          }
+          // cache the mapping in memory to avoid duplicate retrieval
+          metricCache.put(me.getId(), datasets);
+        } else {
+          // retrieve dataset mapping from memory
+          if (metricCache.get(me.getId()) != null)
+            datasets.addAll(metricCache.get(me.getId()));
+        }
+      }
+      dataset2DetectionMap.put(datasets, detectionConfig);
+      for (String dataset : datasets) {
+        if (!datasetRefreshTimeMap.containsKey(dataset)) {
+          DatasetConfigDTO datasetConfig = datasetConfigManager.findByDataset(dataset);
+          datasetRefreshTimeMap.put(dataset, datasetConfig.getLastRefreshTime());
+        }
+      }
+    }
+  }
+
+  private Map<Long, TaskDTO> retrieveRunningDetection() {
+    TaskManager taskManager = DAORegistry.getInstance().getTaskDAO();
+    List<TaskConstants.TaskStatus> statuses = new ArrayList<>();
+    statuses.add(TaskConstants.TaskStatus.WAITING);
+    statuses.add(TaskConstants.TaskStatus.RUNNING);
+    List<TaskDTO> tasks = taskManager.findByStatusesAndTypeWithinDays(statuses, TaskConstants.TaskType.DETECTION,
+        (int) TimeUnit.MILLISECONDS.toDays(ThirdEyeUtils.DETECTION_TASK_MAX_LOOKBACK_WINDOW));
+    Map<Long, TaskDTO> res = new HashMap<>(tasks.size());
+    for (TaskDTO task : tasks) {
+      String[] parts = task.getJobName().split("_");
+      if (parts.length < 2) {
+        LOG.error("Invalid name found {} for task id {}", task.getJobName(), task.getId());
+        continue;
```

Comment:
I would say that keeping it relaxed is better because we may append more information to the `JobName` in the future. If that's the case, we don't need to change this line.